tools/download-srs.sh

```diff
@@ -0,0 +1,225 @@
+    fi
+    
+    # Check g1.point size
+    if ! curl_result=$(curl -f -S -I "${base_url}/g1.point"); then
+        echo "Error: Cannot access ${base_url}/g1.point. The URL may be incorrect or the server is unavailable."
+        return 1
+    fi
+    
+    g1_total_size=$(echo "$curl_result" | grep -i "Content-Length" | awk '{print $2}' | tr -d '\r')
+    
+    if [ -z "$g1_total_size" ]; then
+        echo "Error: Could not determine g1.point file size"
+        return 1
+    fi
+    
+    echo "Total g1.point size: ${g1_total_size} bytes"
+    echo "Total g2.point size: ${g2_total_size} bytes"
+    
+    # Validate that our request sizes are reasonable
+    if [ "$size_bytes" -gt "$g1_total_size" ]; then
+        echo "Error: Requested g1 size ($size_bytes bytes) is larger than the source g1.point file ($g1_total_size bytes)"
+        return 1
+    fi
+    
+    if [ "$g2_size_bytes" -gt "$g2_total_size" ]; then
+        echo "Error: Requested g2 size ($g2_size_bytes bytes) is larger than the source g2.point file ($g2_total_size bytes)"
+        return 1
+    fi
+    
+    # Calculate the start position for g2.trailing.point
+    # Since we want the last G2_SIZE_BYTES of the file
+    local g2_trailing_start=$((g2_total_size - g2_size_bytes))
+    
+    # g2_trailing_start should never be negative due to the validation above,
+    # but if it is somehow negative, we should fail rather than download incorrect data
+    if [ "$g2_trailing_start" -lt 0 ]; then
+        echo "Error: Cannot download g2.trailing.point of size ${g2_size_bytes} bytes - insufficient source data"
+        return 1
+    fi
+    
+    # Download g1.point file
+    if ! download_file "${base_url}/g1.point" "${output_dir}/g1.point" "0-$((size_bytes - 1))" "g1.point (${size_bytes} bytes)"; then
+        return 1
+    fi
+    
+    # Download g2.point file
+    if ! download_file "${base_url}/g2.point" "${output_dir}/g2.point" "0-$((g2_size_bytes - 1))" "g2.point (${g2_size_bytes} bytes)"; then
+        return 1
+    fi
+    
+    # Download g2.trailing.point file
+    if ! download_file "${base_url}/g2.point" "${output_dir}/g2.trailing.point" "${g2_trailing_start}-$((g2_total_size - 1))" "g2.trailing.point (${g2_size_bytes} bytes from the end of g2.point)"; then
+        return 1
+    fi
+    
+    # Generate hashes for verification
+    echo "Generating verification hashes..."
+    
+    # Check if all files exist before generating hashes
+    for file in "${output_dir}/g1.point" "${output_dir}/g2.point" "${output_dir}/g2.trailing.point"; do
+        if [ ! -f "$file" ]; then
+            echo "Error: File $file not found. Cannot generate hashes."
+            return 1
+        fi
+    done
+    
+    # Generate the hash file
+    {
+        echo "# SRS files hashes for blob size ${size_bytes} bytes"
+        echo "# Generated on $(date)"
+        echo "# Format: SHA256 (filename)"
+        echo ""
+        sha256sum "${output_dir}/g1.point" | sed "s|${output_dir}/||"
```

Comment:
Should we store the sha value in the repo?

Reply:
It's included in the README for the "standard" file size.

Are you saying you think it _shouldn't_ be there, or just didn't see that it was?

Reply:
I was thinking something like this https://github.com/Layr-Labs/eigenda-operator-setup/blob/v0.9.0-rc.1/resources/srssha256sums.txt

Reply:
I see I missed that it was in the README.md

Reply:
I'll commit the sha file in addition to leaving it in the readme 👍